### PR TITLE
Searching for things like "World" in "Hello World." returns no results.

### PR DESCRIPTION
**Pull request by oOntheway**
Signed-off-by: Kevin <dl_xq@sina.com>

## Hey, I just made a Pull Request!

In [release-2021-06-10.1](https://github.com/backstage/backstage/releases/tag/release-2021-06-10.1), search results are not accurate.

### Current Behavior

- Searching for things like `World` in `Hello World.`: returns empty.
- Searching for things like `World.` in `Hello World.`: returns match for `World`.

### Expected Behavior

- Searching for things like `World` in `Hello World.`: returns match for `World`.

### @backstage Versions

- app
  - @backstage/plugin-search@0.4.0
- backend
  - @backstage/plugin-search-backend@0.2.0
  - @backstage/plugin-search-backend-node@0.2.0

### Reference

Reference resources: https://github.com/olivernn/lunr.js/blob/master/lib/lunr.js#L35.

```js
var lunr = function (config) {
  var builder = new lunr.Builder

  builder.pipeline.add(
    lunr.trimmer,
    lunr.stopWordFilter,
    lunr.stemmer
  )

  builder.searchPipeline.add(
    lunr.stemmer
  )

  config.call(builder, builder)
  return builder.build()
}
```

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by Fox32** [comment]
Yeah, that is also something I observed. For a full text search, the current lunr configuration is quite picky how you form search terms.

To get this PR accepted, you need to add a changeset. See https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets

**Review by OrkoHunter — APPROVED** [review]
@backstage/techdocs-core Over to you for final review and merge :)

**Comment by Fox32** [comment]
Should `LunrSearchEngine.test.ts` be extended to include a test for this case?

File: .changeset/cyan-drinks-dream.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-search-backend-node': patch
+---
+
+Searching for things like "World" in "Hello World." returns no results.
```
Comment (iamEAP) [inline .changeset/cyan-drinks-dream.md:5]: Recommend something like...

> Improved the quality of free text searches in LunrSearchEngine.

Or similar.  Reason being, this text gets auto-added to `CHANGELOG.md`, so framing it as what was fixed vs. what was broken is helpful for future adopters!

**Review by iamEAP — APPROVED** [review]
Thanks for the contribution @oOntheway!  This'll improve things quite a bit.  Minor note on the changeset if you're going to look at the tests anyway.

File: .changeset/cyan-drinks-dream.md (hunk 1 above)
Comment (oOntheway) [reply .changeset/cyan-drinks-dream.md]: @iamEAP Thanks ! Changeset updated. And it's better if you are going to add the tests. 👍

Comment (Fox32) [reply .changeset/cyan-drinks-dream.md]: I updated the PR to have a test for trimming and one for stemming.

**Comment by iamEAP** [comment]
Dream team!  Congrats on being the first addition to the alpha search platform.

**Pull request merged**